inst/training_202307/202307.qmd

```diff
@@ -0,0 +1,448 @@
+---
+title: "`verdepcheck`"
+author: "Pawel Rucki"
+date: 2023/08/07
+format: revealjs
+---
+
+## Agenda
+
+1. Motivation
+1. Solution
+1. How to use?
+1. How it works?
+1. Discussion
+1. Future work
+
+# Motivation
+
+```{.yaml code-line-numbers="7"}
+Type: Package
+Package: foo.package
+Version: 1.2.3.9000
+Depends:
+    R (>= 3.6)
+Imports:
+    dplyr,
+    (...)
+```
+
+The above specification indicates compatibility with **all** versions of `dplyr` which is very unlikely to be true.
+
+# Motivation
+
+```.default
+Error: object ‘foo’ is not exported by 'namespace:bar'
+```
+
+<br>
+
+```.default
+`foo()` was deprecated in bar 1.0.0.
```

Comment:
this isn't going to be solved by a minimum dependency, and as I mentioned in the PI planning, maximum version dependencies might be technically supported by the DESCRIPTION file spec (honestly the writing r extensions manual just mentions "comparison operator" but the only example they give is <= so I'm skeptical).

Furthermore, even if it technically works in R, I'm very doubtful it would be accepted on CRAN.